src/compress/parallel_gzip.rs

```diff
@@ -0,0 +1,51 @@
+use gzp::ZWriter;
+use std::io::{Result, Write};
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+
+struct UnasyncWriter<W>(W);
+
+impl<W: AsyncWrite + Unpin> Write for UnasyncWriter<W> {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        crate::run::tokio_runtime().block_on(self.0.write(buf))
```

Comment:
i'm actually not fully certain what this does if called from the main thread. did you check if two compression tasks can happen simultaneously if you call

```rust
tokio::join!(compress(in1, out1, pb1), compress(in2, out2, pb2))
```

Reply:
I added a test case which surprisingly passes with a single threaded tokio runtime.